src/Discovery/CollaboraDiscoveryFetcher.php

```diff
@@ -12,47 +12,78 @@
 
 declare(strict_types=1);
 
-namespace Drupal\collabora_online\Cool;
+namespace Drupal\collabora_online\Discovery;
 
 use Drupal\collabora_online\Exception\CollaboraNotAvailableException;
+use Drupal\Component\Datetime\TimeInterface;
+use Drupal\Core\Cache\CacheBackendInterface;
 use Drupal\Core\Config\ConfigFactoryInterface;
-use Drupal\Core\Logger\LoggerChannelInterface;
+use Drupal\Core\Config\ImmutableConfig;
 use GuzzleHttp\ClientInterface;
 use GuzzleHttp\RequestOptions;
 use Psr\Http\Client\ClientExceptionInterface;
+use Psr\Log\LoggerInterface;
 use Symfony\Component\DependencyInjection\Attribute\Autowire;
 
 /**
  * Service to load the discovery.xml from the Collabora server.
  */
 class CollaboraDiscoveryFetcher implements CollaboraDiscoveryFetcherInterface {
```

Comment:
@brummbar 
I have an alternative branch (locally) where I merge/combine the fetcher into the loader.